pkg/client/client.go

```diff
@@ -68,6 +69,7 @@ type Client struct {
 	Discovery     discovery.DiscoveryInterface
 	PrometheusCli monclientv1.MonitoringV1Interface
 	DeploymentCli appsv1.AppsV1Interface
+	KubernetesCli kubernetes.Interface
```

Comment:
I might be missing something, but why this? The client itself is a Kubernetes client, the others are extensions. We already have the `Namespace` interface that's suppose to work on the same way as the `Pod` you introduced. Sorry, I haven't enough time to go through everything yet. Could you please clarify? Thanks!!

Reply:
Well, the problem is that log cannot be retrieved by CRUD on Kubernetes objects as log isn't a usual object.
It is retrieved as a subresource [1] using client. I didn't find any better way to retrieve it. Do you have any idea how can this be improved?

[1] https://github.com/alankritr/kubernative_lib/blob/4e35750abcfc7848719b84dd04346488fc2aa9fa/staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod_expansion.go#L44

Reply:
Should we use k8s.io/client-go (currently used) dependency or github.com/kubernetes/client-go?
I can see that we already use github.com/openshift/client-go

Reply:
the openshift one is to query specifics APIs from Openshift such as ImageStream. Please stick with k8s.io if possible.

We could rename the `Kubernetes` API to `KubernetesExtension` instead, so won't cause confusion in the future.

Reply:
Done, please take a look